manifests/manifests.go

```diff
@@ -2,68 +2,73 @@ package manifests
 
 import (
 	"bytes"
+	"embed"
 	_ "embed"
 	"fmt"
+	"io/fs"
+	"path/filepath"
 	"text/template"
+)
 
-	"github.com/pkg/errors"
+const (
+	wegoManifestsDir = "wego-app"
 )
 
-var Manifests [][]byte
+var (
+	Manifests [][]byte
+	//go:embed crds/wego.weave.works_apps.yaml
+	AppCRD []byte
+	//go:embed wego-app/*
+	wegoAppTemplates embed.FS
+)
 
-//go:embed crds/wego.weave.works_apps.yaml
-var AppCRD []byte
+type WegoAppParams struct {
+	Version   string
+	Namespace string
+}
 
-//go:embed wego-app/deployment.yaml
-var WegoAppDeployment []byte
+// GenerateWegoManifests generates wego-app manifest from a template
+func GenerateWegoAppManifests(params WegoAppParams) ([][]byte, error) {
+	manifests := [][]byte{}
 
-type deploymentParameters struct {
-	Version string
-}
+	templates, _ := fs.ReadDir(wegoAppTemplates, wegoManifestsDir)
+	for _, template := range templates {
+		tplName := template.Name()
 
-var errInjectingValuesToTemplate = errors.New("error injecting values to template")
+		data, err := fs.ReadFile(wegoAppTemplates, filepath.Join(wegoManifestsDir, tplName))
+		if err != nil {
+			return nil, fmt.Errorf("failed reading template %s: %w", tplName, err)
+		}
 
-// GenerateWegoAppDeploymentManifest generates wego-app deployment manifest from a template
-func GenerateWegoAppDeploymentManifest(version string) ([]byte, error) {
-	deploymentValues := deploymentParameters{Version: version}
+		manifest, err := executeTemplate(tplName, string(data), params)
+		if err != nil {
+			return nil, fmt.Errorf("failed executing template: %s: %w", tplName, err)
+		}
 
-	template := template.New("DeploymentTemplate")
+		manifests = append(manifests, manifest)
+	}
 
-	var err error
+	return manifests, nil
+}
 
-	template, err = template.Parse(string(WegoAppDeployment))
+func executeTemplate(name string, tplData string, params WegoAppParams) ([]byte, error) {
+	template, err := template.New(name).Parse(tplData)
 	if err != nil {
-		return nil, fmt.Errorf("error parsing template %w", err)
+		return nil, fmt.Errorf("error parsing template %s: %w", name, err)
 	}
 
-	deploymentYaml := &bytes.Buffer{}
+	yaml := &bytes.Buffer{}
 
-	err = template.Execute(deploymentYaml, deploymentValues)
+	err = template.Execute(yaml, params)
 	if err != nil {
-		return nil, fmt.Errorf("%s %w", errInjectingValuesToTemplate, err)
+		return nil, fmt.Errorf("error injecting values to template: %w", err)
 	}
 
-	return deploymentYaml.Bytes(), nil
+	return yaml.Bytes(), nil
 }
 
-//go:embed wego-app/service-account.yaml
-var WegoAppServiceAccount []byte
-
-//go:embed wego-app/service.yaml
-var WegoAppService []byte
-
-//go:embed wego-app/role.yaml
-var WegoAppRole []byte
-
-//go:embed wego-app/role-binding.yaml
-var WegoAppRoleBinding []byte
-
 func init() {
 	Manifests = [][]byte{
```

Comment:
Do we think this will ever have more than one manifest in it any more? If not, maybe this should just be an "AppCRDManifest" variable :thinking:

Reply:
yep, I'll remove it